src/main.rs

```diff
@@ -72,27 +77,33 @@ fn create_project(paths: &constants::Paths, shell_nix: NixFile) -> Result<Projec
 }
 
 /// Run the main function of the relevant command.
-fn run_command(opts: Arguments) -> OpResult {
+fn run_command(log: slog::Logger, opts: Arguments) -> OpResult {
     let paths = lorri::ops::get_paths()?;
+    let nix_file = match &opts.command {
+        Command::Info(opts) => Some(&opts.nix_file),
+        Command::Direnv(opts) => Some(&opts.nix_file),
+        Command::Watch(opts) => Some(&opts.nix_file),
+        Command::Ping_(opts) => Some(&opts.nix_file),
+        _ => None,
+    };
+    let (log, project) = if nix_file.is_none() {
+        (log, None)
+    } else {
+        let project = get_shell_nix(nix_file.unwrap()).and_then(|sn| create_project(&paths, sn))?;
+        let log = log.new(slog::o!("root" => project.nix_file.clone()));
+        (log, Some(project))
+    };
```

Comment:
What would you think about this alternative take on this code's path:

1. adding to cli.rs:


```rust
impl Command {
    pub fn nix_file(&self) -> Option<&PathBuf> {
        match self {
            Command::Info(opts) => Some(&opts.nix_file),
            Command::Direnv(opts) => Some(&opts.nix_file),
            Command::Watch(opts) => Some(&opts.nix_file),
            Command::Ping_(opts) => Some(&opts.nix_file),
            _ => None,
        }
    }
}
```

and then this part of this function becomes:

```rust
    let paths = lorri::ops::get_paths()?;

    let project: Option<Project> = opts
        .command
        .nix_file()
        .map(|nix_file: &PathBuf| get_shell_nix(nix_file))
        .transpose()?
        .map(|nix_file: NixFile| create_project(&paths, nix_file))
        .transpose()?;

    let logged_root: Option<NixFile> = project.as_ref().map(|p| p.nix_file.clone());

    let log = log.new(slog::o!("root" => logged_root));
```

and as a side-effect, I think the following code can be improved a bit too -- removing `unwrap`s and what-not.

Reply:
Thanks for the suggestion! This is a big improvement in readability. I wonder if we can go further. I'll have a think about this.

Reply:
I've pushed a change that uses more explicit setup code in `main.rs`. It's not the prettiest code in the world, but its purpose is reasonably clear I believe. Let me know what you think!